src/SDKs/SqlManagement/Sql.Tests/SqlDatabaseAgentScenarioTests.cs

```diff
@@ -0,0 +1,601 @@
+﻿using Microsoft.Azure.Management.ResourceManager;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.Azure.Management.Sql;
+using Microsoft.Azure.Management.Sql.Models;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Rest.Azure;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Sql.Tests
+{
+    public class SqlDatabaseAgentScenarioTests
+    {
+        private string _subscriptionId;
+
+        public SqlDatabaseAgentScenarioTests()
+        {
+            // Sort of a hacky way to grab the subscription id from the environment variables, but given that developer set their
+            // TEST_CSM_ORGID_AUTHENTICATION environment variable in this format then:
+            // SubscriptionId={SubId};ServicePrincipal={clientId};ServicePrincipalSecret={clientSecret};AADTenant={tenantId};Environment={env};HttpRecorderMode=Record;
+            // The below string split and getting the {SubId} should work.
+            this._subscriptionId = Environment.GetEnvironmentVariables()["TEST_CSM_ORGID_AUTHENTICATION"].ToString().Split(';')[0].Split('=')[1];
```

Comment:
I think this won't work in test playback, but you can use `sqlManagementClient.SubscriptionId` instead.

Reply:
Won't need it since we can just use credential.Id and targetGroup.Id :)